File: system-design/oop-patterns/02-observer-pattern/full/ts/weather-station/WindowDisplay.ts
import type { IObserver } from './IObserver';
import type { WeatherStation } from './WeatherStation';

export class WindowDisplay implements IObserver {
  constructor(private station: WeatherStation) {}

  update(): void {
    console.log(`Window: ${this.station.getTemperature()} degrees outside.`);
  }
}

File: system-design/oop-patterns/02-observer-pattern/full/java/weatherstation/WindowDisplay.java
package weatherstation;

public class WindowDisplay implements IObserver {
    private final WeatherStation station;

    public WindowDisplay(WeatherStation station) {
        this.station = station;
    }

    public void update() {
        System.out.println("Window: " + station.getTemperature() + " degrees outside.");
    }
}
